compiler/luci/pass/src/DecomposeHardSwishPass.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/DecomposeHardSwishPass.h"
+
+#include "helpers/NodeFiller.h"
+#include "helpers/TypeMapper.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+namespace
+{
+/**
+ *  BEFORE
+ *        [CircleNode]
+ *              |
+ *              |
+ *      [CircleHardSwish]
+ *              |
+ *              |
+ *        [CircleNode]
+ *
+ *
+ *  AFTER
+ *
+ *      [CircleNode]  [CircleConst]
+ *          |    \       /
+ *          |     \     /
+ *          |   [CircleAdd]
+ *          |        |
+ *          |        |
+ *          \  [CircleRelu6] [CircleConst]
+ *           \        \        /
+ *            \        \      /
+ *             \      [CircleMul]
+ *              \       /
+ *               \     /
+ *             [CircleMul]
+ *                  |
+ *                  |
+ *             [CircleNode]
+ *
+ */
+template <loco::DataType DT> bool decompose_hardswish(luci::CircleHardSwish *hardswish)
+{
+  if (not hardswish)
+    return false;
+
+  if (hardswish->dtype() != DT)
+    return false;
+
+  auto g = hardswish->graph();
+
+  auto name = hardswish->name();
+  assert(name.length() > 0);
+
+  // Create a const for CircleAdd operation
+  auto add_const = g->nodes()->create<luci::CircleConst>();
+  add_const->shape({}); // scalar
+  add_const->dtype(DT);
+  add_const->rank(0);
+  add_const->size<DT>(1);
+  add_const->at<DT>(0) = 3.;
+  add_const->name("add_const");
```

Comment:
how about `->name(name + "/add_const")` ?
this is to specify this constant is related with source HardSwish node.

Reply:
It would make more sense to do so.
Thanks for the suggestion!

Reply:
applied =)